cogs5e/models/initiative/combat.py

```diff
@@ -279,16 +279,17 @@ def get_group(self, name, create=None, strict=True):
         :param strict: Whether group name must be a full case insensitive match.
         :return: The combatant group.
         """
-        if name in self._combatant_id_map and isinstance(self._combatant_id_map[name], CombatantGroup):
-            return self._combatant_id_map[name]
-        if strict:
-            grp = next((g for g in self.get_groups() if g.name.lower() == name.lower()), None)
-        else:
-            grp = next((g for g in self.get_groups() if name.lower() in g.name.lower()), None)
+        if name:  # catch no group without having to loop
```

Comment:
What case spurred this change, out of curiosity?

Reply:
Also, to prevent having to indent the rest of the method, it's cleaner to do 
```py
if not name:
    return None
```
(in fact, I think this raises an UnboundLocalError, since `grp` is undefined when it reaches the return)

Reply:
Although, an empty name might be better off raising something, since it's not really a valid group name anyway, and just returning None might have unexpected effects when `create` is set

Reply:
Although the init system does let group names be empty so it might be better to just eat the loops here

Reply:
Oh, I see, `name` might be `None` from `Combatant.get_group_name` - I think that check should be done in the combatant's method, then